subprojects/docs/src/docs/userguide/api/kotlin_dsl.adoc

```diff
@@ -734,86 +734,9 @@ In future releases, we will work on improving the IDE experience and remove thes
 
 
 [[sec:kotlin-dsl_plugin]]
-== The Kotlin DSL Plugin
+== Writing plugins in Kotlin
 
-The Kotlin DSL Plugin provides a convenient way to develop Kotlin-based projects that contribute build logic.
-That includes <<organizing_gradle_projects#sec:build_sources,buildSrc projects>>, <<composite_builds#composite_builds,included builds>> and <<custom_plugins#custom_plugins,Gradle plugins>>.
-
-The plugin achieves this by doing the following:
-
- * Applies the link:https://kotlinlang.org/docs/reference/using-gradle.html#targeting-the-jvm[Kotlin Plugin], which adds support for compiling Kotlin source files.
- * Adds the `kotlin-stdlib-jdk8`, `kotlin-reflect` and `gradleKotlinDsl()` dependencies to the `compileOnly` and `testImplementation` configurations, which allows you to make use of those Kotlin libraries and the Gradle API in your Kotlin code.
- * Configures the Kotlin compiler with the same settings that are used for Kotlin DSL scripts, ensuring consistency between your build logic and those scripts:
- ** adds <<kotlin_dsl#sec:kotlin_compiler_arguments, Kotlin compiler arguments>>,
- ** registers the link:https://kotlinlang.org/docs/sam-with-receiver-plugin.html[SAM-with-receiver Kotlin compiler plugin].
- * Enables support for <<custom_plugins#sec:precompiled_plugins,precompiled script plugins>>.
-
-[CAUTION]
-.Avoid specifying a version for the `kotlin-dsl` plugin
-====
-Each Gradle release is meant to be used with a specific version of the `kotlin-dsl` plugin and compatibility between arbitrary Gradle releases and `kotlin-dsl` plugin versions is not guaranteed. Using an unexpected version of the `kotlin-dsl` plugin in a build will emit a warning and can cause hard to diagnose problems.
-====
-
-This is the basic configuration you need to use the plugin:
-
-.Applying the Kotlin DSL Plugin to a `buildSrc` project
-====
-include::sample[dir="snippets/kotlinDsl/kotlinDslPlugin/kotlin",files="buildSrc/build.gradle.kts[tags=apply]"]
-====
-
-The Kotlin DSL Plugin leverages <<toolchains#toolchains,Java Toolchains>>.
-By default the code will target Java 8.
-You can change that by defining a Java toolchain to be used by the project:
-
-.Changing the JVM target using toolchains
-====
-include::sample[dir="samples/java/jvm-multi-project-with-toolchains/kotlin/",files="buildSrc/src/main/kotlin/myproject.java-conventions.gradle.kts[tags=toolchain]"]
-include::sample[dir="samples/java/jvm-multi-project-with-toolchains/groovy/",files="buildSrc/src/main/groovy/myproject.java-conventions.gradle[tags=toolchain]"]
-====
-
-[[sec:kotlin]]
```

Comment:
Much more seems to be removed from here than is added back in the new page... Maybe removal should have stopped at this line?

Reply:
I've put it back. I'd prefer `Kotlin compiler arguments` to be left out: As a Kotlin DSL user, I shouldn't be exposed too much to that. But no strong feeling on this.